tests/CacheCompat/CommonCache.Test.AdalV3/Program.cs

```diff
@@ -2,6 +2,8 @@
 // Licensed under the MIT License.
 
 using System;
```

Comment:
seems like all the tests are doing a lot of the same stuff... perhaps some of that stuff could be in the base (user looping and adding stuff to the results?

Reply:
yeah, there's possible room for refactoring here but there are subtle differences along the way we'd have to build some type of template execution base class so we can hook in all of the right places for different cache behavior or acquire token methods since all of the APIs are subtly different.